servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -952,19 +993,39 @@ private TypeSpec newClientToBlockingClientClassSpec(final State state,
         return typeSpecBuilder.build();
     }
 
+    /**
+     * Adds the service interface, either async or blocking, which extends all of the service RPC interfaces
+     *
+     * @param state The generator state
+     * @param blocking If true then add the interface for blocking service otherwise add async service interface
+     * @return The generated service interface
+     */
     private TypeSpec newServiceInterfaceSpec(final State state, final boolean blocking) {
-        final String name = context.deconflictJavaTypeName((blocking ? Blocking : "") +
-                sanitizeIdentifier(state.serviceProto.getName(), false) + "Service");
+        final ClassName serviceClass = blocking ? state.blockingServiceClass : state.serviceClass;
+        final String name = serviceClass.simpleName();
 
         final TypeSpec.Builder interfaceSpecBuilder = interfaceBuilder(name)
                 .addModifiers(PUBLIC)
+                .addSuperinterface(GrpcBindableService)
```

Comment:
we should also add the generic type arguments

Reply:
Ooops, forgot them